### DISCUSSION TITLE
get all plugin params (and defaults) for a specific release as javascript object or json

### DISCUSSION THREAD

**Discussion by kschuler** [jspsych/jsPsych#1205] [Q&A]
My lab loves jsPsych. We've developed a builder for jsPsych experiments to make it go a little smoother for new researchers. Below is a screenshot of the playground where our research assistants can practice setting parameters for different plugins.

To do this, we enter each plugin by hand into a database. The plugins name, docs link, description, and a json of the parameters that plugin accepts including its default values. Right now, my lab manager is tasked with entering each by hand, but we are wondering if you happen to have a javascript object or json that already has the plugin parameter values that we could use. We plan to release our builder (open to everyone) when we are done. :)

![Screen Shot 2020-11-20 at 8 37 40 AM](https://user-images.githubusercontent.com/6099948/99806800-88397180-2b0c-11eb-8670-88c0fd9e1eee.png)

**Comment by jodeleeuw** [comment]
Hi Kathryn,

This looks very cool! Once you've got this ready for the public it would be really nice to link to these playground from the main docs.

A few years ago we were working on our own experiment builder, and so we added `plugin.info` to each of the plugins. Here's the relevant bit from the html-keyboard-response plugin, for example:

https://github.com/jspsych/jsPsych/blob/005772d78b8cb6e573f668e67df54c8c8d525cb5/plugins/jspsych-html-keyboard-response.js#L16-L59

When we were working on our builder (which has been de-prioritized for now to work more on the code base) we would load in the plugin files and then extract the `plugin.info` object from each to get this information.

If your team needs more information than is currently in the object I'd be open to considering expanding the information that's contained in it, especially if someone in your group is willing to go through and add/fix up all of the plugin files.

**Reply by kschuler** [reply]
Thanks very much! I hadn't noticed these were available inside the plugin, this is a big help. :) 

I'll for sure send a link to the playground when it's ready for public use.
